### DISCUSSION TITLE
How to bundle TypeScript declaration files of Vue project?

### DISCUSSION THREAD

**Discussion by Zhang-Wei-666** [web-infra-dev/rslib#505] [Q&A] [answered]
我想把使用 tsc 打包出的多个分散的 .d.ts 类型文件, 打包为一个 .d.ts 类型文件

---

I want to combine multiple scattered .d.ts type files generated by tsc into a single .d.ts type file.

**Comment by Timeless0911** [comment]
https://lib.rsbuild.dev/guide/advanced/dts#how-to-generate-dts-in-rslib

**Comment by Zhang-Wei-666** [comment]
@Timeless0911  似乎并不行, 复现在这里:

https://github.com/Zhang-Wei-666/rslib-bug-report/tree/4

以下是截图

![PixPin_2024-12-02_14-09-10](https://github.com/user-attachments/assets/b4319a0e-7402-4214-b267-7cb6ae8a0841)

**Reply by Timeless0911** [reply]
![image](https://github.com/user-attachments/assets/d245b90f-768b-4227-94ba-9355917cde28)

Why you source code have double `src` folder and is already `.d.ts` instead of `.ts`. And if you are vue project, bundleless mode and DTS are both not available yet.

**Reply by Zhang-Wei-666** [reply]
这是用 vue-tsc 打包出的类型, 但是 vue-tsc 打包出来后, 第三方类库中的类型并没有被打包进来, 我就想着用 rslib 把类型文件再打包一次

---

This is the type packaged with vue-tsc, but after packaging with vue-tsc, the types from third-party libraries were not included. I thought about using rslib to package the type files again.



![PixPin_2024-12-02_14-22-38](https://github.com/user-attachments/assets/cfc2d9ab-6229-4dd4-a6fd-128e399791a3)

**Reply by Timeless0911** [reply, marked as answer]
The DTS bundle feature in Rslib is generated by Typescript Compiler API and then bundled by api-extractor. 

If you want to bundle Vue DTS files, you can use other tools. see https://github.com/timocov/dts-bundle-generator/discussions/68

**Reply by Zhang-Wei-666** [reply]
稳的, 已经实现了, 没问题

https://github.com/MoomFE/mixte/blob/e7831ba19e61021d6d1e1a7a18d48e2c8873ffbe/scripts/build-utils.ts#L58-L84
